src/generator/doc_repo.cr

```diff
@@ -124,6 +124,15 @@ module Generator
     def crystallize_doc(doc : String) : String
       crystallized_doc = doc
       crystallized_doc = crystallized_doc.gsub(/```([a-zA-Z]+)\n/, "\n\nWARNING: **⚠️ The following code is in \\1 ⚠️**\n```\\1\n")
+
+      crystallized_doc = crystallized_doc.gsub(/%(\w+)/) do |_, match|
+        types = {null: "Nil"}
+        downcased_type = match[-1].downcase
+
+        crystallized_type = types.fetch(downcased_type, downcased_type)
+        "`#{crystallized_type}`"
```

Comment:
I didn't measure anything :sweat_smile: , but probably using a small case/when is faster because we don't need to allocate memory downcasing the strings, also is better to use a more specific regex, so we know that the block is executed only when we care about.

```suggestion
      crystallized_doc = crystallized_doc.gsub(/%(TRUE|FALSE|NULL)/) do |_, match|
        case match[1]
        when "TRUE" then "`true`"
        when "FALSE" then "`false`"
        when "NULL" then "`nil`"
        end
```